tests/vector_tools/interpolate_distributed_01.cc

```diff
@@ -0,0 +1,80 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2018 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+
+// Tests if proper checks are run when interpolating on distributed vectors
+
+#include <deal.II/base/function.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/mpi.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/fe/mapping_q.h>
+
+#include <deal.II/grid/filtered_iterator.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/grid_tools.h>
+#include <deal.II/grid/grid_tools_cache.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/tria.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+#include "../tests.h"
+
+using namespace dealii;
+
+template <int dim>
+void
+test_interpolating_distributed(unsigned int ref_cube)
+{
+  MPI_Comm     mpi_communicator = MPI_COMM_WORLD;
+  unsigned int n_procs = Utilities::MPI::n_mpi_processes(mpi_communicator);
+
+  deallog << "Testing for dim = " << dim << " on " << n_procs << " processes"
+          << std::endl;
+  deallog << "Cube refinements: " << ref_cube << std::endl;
+
+  FE_Q<dim> fe(1);
+  // Implementing the constant function
+  Functions::ConstantFunction<dim> f_constant(1.0);
+
+  // Creating a distributed triangulation
+
+  parallel::distributed::Triangulation<dim> cube_d(mpi_communicator);
+  GridGenerator::hyper_cube(cube_d);
+  cube_d.refine_global(ref_cube);
+  deallog << "Initializing the dof handler" << std::endl;
+  DoFHandler<dim> dof_handler_d(cube_d);
+  dof_handler_d.distribute_dofs(fe);
+
+  deallog << "Interpolating data" << std::endl;
+  LinearAlgebra::distributed::Vector<double> data_d(dof_handler_d.n_dofs());
```

Comment:
I have not looked into this more closely yet, but shouldn't you provide a partitioning at this point? If you want to use the vector in parallel, you would specify the locally owned range of the DoFHandler and the relevant dofs as ghost (for example). The tests in themselves seem to make sense, but I do not understand the test (yet).